src/XRP/xrp-payment.ts

```diff
@@ -0,0 +1,79 @@
+import { Payment } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import XRPCurrencyAmount from './xrp-currency-amount'
+import XRPPath from './xrp-path'
+
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPPayment {
+  public static from(payment: Payment): XRPPayment | undefined {
+    let amount
+    const paymentAmountValue = payment.getAmount()?.getValue()
+    if (paymentAmountValue) {
+      amount = XRPCurrencyAmount.from(paymentAmountValue)
+      if (!amount) {
+        return undefined // amount is required
+      }
+    } else {
+      return undefined
+    }
+    const destination = payment
+      .getDestination()
+      ?.getValue()
+      ?.getAddress()
+    if (!destination) {
+      return undefined // destination is required
+    }
+    const destinationTag = payment.getDestinationTag()?.getValue()
+    // If the deliverMin field is set, it must be able to be transformed into a XRPCurrencyAmount.
+    let deliverMin
+    if (payment.hasDeliverMin()) {
+      const paymentDeliverMinValue = payment.getDeliverMin()?.getValue()
+      if (paymentDeliverMinValue) {
+        deliverMin = XRPCurrencyAmount.from(paymentDeliverMinValue)
+        if (!deliverMin) {
+          return undefined
+        }
+      }
+    } else {
+      deliverMin = undefined
+    }
```

Comment:
Again, what is `XRPCurrencyAmount.from(undefined)`?

Maybe this could look like this:
```ts
    // If the deliverMin field is set, it must be able to be transformed into a XRPCurrencyAmount.
    let deliverMin = undefined
    if (payment.hasDeliverMin()) {
      const paymentDeliverMinValue = payment.getDeliverMin()?.getValue()
      deliverMin = XRPCurrencyAmount.from(paymentDeliverMinValue)
      if (!deliverMin) {
        return undefined
      }
    }
```

We could even make this shorter (depending on return values):
```ts
    // If the deliverMin field is set, it must be able to be transformed into a XRPCurrencyAmount.
    const paymentDeliverMinValue = payment.getDeliverMin()?.getValue()
    const deliverMin = XRPCurrencyAmount.from(paymentDeliverMinValue)
    if (payment.hasDeliverMin() && !deliverMin) {
      return undefined
    }
```

Reply:
same question as above... Keefer do you want to weigh in on whether we should update the `.from` initializers so that we can feed them `undefined` values and use their return to make things more succinct?

Reply:
See comment above about `amount`.